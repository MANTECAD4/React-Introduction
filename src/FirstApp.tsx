
type Props = {
    title:string,
    subTitle?:string,
    name?:string,
}

export const FirstApp = ( {title, subTitle='No subtitle', name='No name' }:Props ) => {
    return (
        // <Fragment>
        //     <h1>Ã‘aÃ±el's First App ğŸ¤¯ğŸ¤ª</h1>
        //     <p>Holaa</p>
        // </Fragment>
        <>
            <h1 data-testid="test-title">{title} </h1>
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <p>{name}'s First App ğŸ¤¯ğŸ¤ª</p>
        </>

    )
}