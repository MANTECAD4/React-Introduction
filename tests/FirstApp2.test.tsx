import { render, screen } from "@testing-library/react";
import { FirstApp } from '../src/FirstApp.tsx';
import React from 'react';

describe(`Tests on "FirstApp" component`, () => { 
    
    const title:string = "First App component"; 
    const subTitle:string = "Basic Component";
    const name:string = 'Daniel';
    
    test('FirstApp component should match a snapshot', () => { 

        const { container } = render(
            <FirstApp title={ title } subTitle={ subTitle } name={ name }
        />);
        expect(container).toMatchSnapshot();

     });
    
     test('FirstApp component should have the title "First App component"', () => { 
        
        render(
            <FirstApp 
                title={ title } 
                subTitle={ subTitle } 
                name={ name }
            />);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
        
    });
    
     test('FirstApp component should have the title inside an h1 element', () => { 
        
        render(
            <FirstApp 
                title={ title } 
                subTitle={ subTitle } 
                name={ name }
            />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
        
    });
    
    
    test('FirstApp component should have two h2 element for the subTitle', () => { 
        
        render(
            <FirstApp 
                title={ title } 
                subTitle={ subTitle } 
                name={ name }
            />);
        
        expect(screen.getAllByText(subTitle).length).toBe(2);
  
    });    

 });