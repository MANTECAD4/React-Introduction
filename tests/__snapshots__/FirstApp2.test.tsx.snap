// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on "FirstApp" component FirstApp component should match a snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    First App component
     
  </h1>
  <h2>
    Basic Component
  </h2>
  <h2>
    Basic Component
  </h2>
  <p>
    Daniel
    's First App ðŸ¤¯ðŸ¤ª
  </p>
</div>
`;
